### PR TITLE
feat: Handle protected identies

### PR DESCRIPTION
**Pull request by andypattenden**
# Description

Adds handling of protected identities.

There is no flag in a response for a protected identity so we have to check certain fields for values. This means that the implementation is fragile should we change service provider or the fields could genuinely have no value.

## Type of change

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Unit tests have been added.

To test in the Q/B app, you will need to npm link this package.

1. Proceed through the Q&B to the personnummer entry.
1. Enter a protected identity number, e.g. 194203052222
1. Click search
1. The message should be displayed that the identity is protected.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] Designer has reviewed the change

### CODE REVIEW

File: src/personnummer.vue [hunk 1]
```diff
@@ -219,6 +227,19 @@ export default {
...
     },
+
+    /**
+     * Checks the returned data for a protected identity
+     *
+     * @param {Object} personData the data to check
+     * @returns {boolean} whether the identity is protected or not
+     */
+    isProtectedIdentity(personData) {
+      const keysToCheck = ['firstName', 'lastName', 'street', 'zipCode', 'city']
```
Comment (shaunkrd) [inline src/personnummer.vue]: Are we confident enough that non-protected identity users will always have all of these in their data? I know there's no docs to follow on this so we have to guess really

Comment (andypattenden) [reply src/personnummer.vue]: Not 100% confident but the above fields are the ones required by the Policy Schema. If any of those are blank, we should fail.

I think there is potentially a more elegant way of handling this so will raise an issue for it.

Comment (andypattenden) [reply src/personnummer.vue]: Raised issue for this in #12

**Comment by andypattenden** [comment]
Closed this in favour of #13

**Pull request closed without merging**